src/processors/modulation/scanner_vibrato/ScannerVibrato.cpp

```diff
@@ -131,25 +136,36 @@ void ScannerVibrato::processAudio (AudioBuffer<float>& buffer)
 
     if (inputsConnected.contains (AudioInput))
     {
+        const auto stereoMode = stereoParam->get();
+
         // generate mod mix arrays
         auto** modMixData = modsMixBuffer.getArrayOfWritePointers();
         FloatVectorOperations::add (modMixData[0], modOutBuffer.getReadPointer (0), 1.0f, numSamples);
         FloatVectorOperations::multiply (modMixData[0], depthParam.getSmoothedBuffer(), numSamples); // this also multiplies the signal by 0.5
-        //        FloatVectorOperations::multiply (modMixData[0], depthParam.getSmoothedBuffer(), modOutBuffer.getReadPointer (0), numSamples); // this also multiplies the signal by 0.5
+
         for (int i = ScannerVibratoWDF::numTaps - 1; i >= 0; --i)
             tapMixTable[i].process (modMixData[0], modMixData[i], numSamples);
 
-        const auto& audioInBuffer = getInputBuffer (AudioInput);
-        const auto numChannels = audioInBuffer.getNumChannels();
-        audioOutBuffer.setSize (numChannels, numSamples, false, false, true);
+        auto& audioInBuffer = getInputBuffer (AudioInput);
+        auto numInChannels = audioInBuffer.getNumChannels();
+        const auto numOutChannels = stereoMode ? 2 : numInChannels;
+
+        if (stereoMode && numInChannels == 1)
```

Comment:
I was having trouble with the mixer when the stereo effect was on, because the input buffer was mono and the output buffer was stereo. I thought having some sort of corrective step was useful to put the input buffer in stereo so the mix would work correctly

Reply:
Yeah, that is tricky. This approach seems good, but I usually try to avoid touching the input buffer if we're doing out-of-place processing, just because other modules might need to use that buffer later.

What if we do something like this:
```cpp
auto& audioInBuffer = getInputBuffer (AudioInput);
const auto numInChannels = audioInBuffer.getNumChannels();
const auto numOutChannels = stereoMode ? 2 : numInChannels;
audioOutBuffer.setSize (numOutChannels, numSamples, false, false, true);
mixer.setWetMixProportion (*mixParam);

if (stereoMode && numInChannels == 1)
{
    for (int ch = 0; ch < numOutChannels; ++ch)
         audioOutBuffer.copyFrom (ch, 0, audioInBuffer, ch % numInChannels, 0, numSamples);

    mixer.pushDrySamples (audioOutBuffer);
}
else
{
     mixer.pushDrySamples (audioInBuffer);
}
```